internal/datastore/postgres/postgres.go

```diff
@@ -378,16 +388,41 @@ func (pgd *pgDatastore) Features(ctx context.Context) (*datastore.Features, erro
 
 func buildLivingObjectFilterForRevision(revision datastore.Revision) queryFilterer {
 	return func(original sq.SelectBuilder) sq.SelectBuilder {
-		return original.Where(sq.LtOrEq{colCreatedTxn: transactionFromRevision(revision)}).
-			Where(sq.Or{
-				sq.Eq{colDeletedTxn: liveDeletedTxnID},
-				sq.Gt{colDeletedTxn: revision},
-			})
+		txID := transactionFromRevision(revision)
+
+		createdBeforeTXN := sq.Expr(fmt.Sprintf(
```

Comment:
do we need to `Sprintf` the arguments of the expression that do not change during the lifetime of the process each time? `colSnapshot`, `tableTransaction` and `colXID`?

Reply:
I thought about that, but it would have required double encoding in a format string, which.... yeesh.

Reply:
I guess it's not a big deal, just those tiny things add up. As I did profiling, putting I/O wait aside, generating SQL with `squirrel` was a visible chunk of the CPU time.